gapis/api/service.proto

```diff
@@ -238,3 +238,127 @@ message CubemapLevel {
 	image.Info negative_z = 5;
 	image.Info positive_z = 6;
 }
+
+message Metrics {
+  oneof metrics {
+    MemoryBreakdown memory_breakdown = 1;
+  }
+}
+
+// The description of the memory layout of the API state
+message MemoryBreakdown {
+  // The API ID used for this call.
+  path.API api = 1;
+  // The memory allocations that are currently in use.
+  repeated MemoryAllocation allocations = 2;
+  // The index to get the constants names for MemoryAllocation `flags`.
+  // A value of -1 indicates no flag names should be fetched.
+  int32 allocation_flags_index = 3;
+}
+
+// A single memory allocation
+message MemoryAllocation {
+  // The device this allocation was made on
+  uint64 device = 1;
+  // The memory type this allocation is from
+  uint32 memory_type = 2;
+  // Flags describing properties of the allocation, e.g. whether it's device
+  // local, whether it's host visible, etc.
+  uint32 flags = 3;
```

Comment:
Part of the goal with this RPC interface is that the client should never have to have intimate knowledge of the actual graphics API. This breaks the abstraction.
I can see two ways of solving this:
1. Have a repeated string field of API flags as names.
2. Making another message with fields like `IsHostVisible`, etc. However this makes it hard to reproduce the true vulkan names.

Reply:
It was originally a repeated string field of API flag names, but @AWoloszyn suggested using `uint32` along with the `allocation_flags_index` field to fetch the flag names from `ConstantSets`

Reply:
Ah okay, please can you document that the flags relate to bitfields in the constant sets then?
This was not obvious.

Reply:
Good point, sure.